### PR TITLE
Fix: fix top_p type hint

### LINKED ISSUE #15341: _OllamaCommon contains top_p with int-restriction
### System Info

LangChain 0.0.353

### Who can help?

@hwchase17
@agola11

### Information

- [ ] The official example notebooks/scripts
- [ ] My own modified scripts

### Related Components

- [X] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

Cannot set top_p to useful values via ChatOllama(top_p=0.3), is reduced to value 0 because it's an int:

top_p: Optional[int] = None

top_p must be float.

### Expected behavior

top_p must be a float, 0.3 should appear in ollama log.

### PR DESCRIPTION
**Pull request by chyroc**
fix: https://github.com/langchain-ai/langchain/issues/15341

@efriis

### CODE REVIEW

File: libs/partners/google-genai/langchain_google_genai/chat_models.py [hunk 1]
```diff
@@ -443,7 +443,7 @@ class ChatGoogleGenerativeAI(BaseChatModel):
     top_k: Optional[int] = None
     """Decode using top-k sampling: consider the set of top_k most probable tokens.
        Must be positive."""
-    top_p: Optional[int] = None
+    top_p: Optional[float] = None
```
Comment (chyroc) [inline libs/partners/google-genai/langchain_google_genai/chat_models.py:446]: https://medium.com/@mryklief21/guide-to-gemini-pro-for-python-563811b3086e
image

File: libs/community/langchain_community/llms/nlpcloud.py [hunk 2]
```diff
@@ -38,7 +38,7 @@ class NLPCloud(LLM):
     """Whether or not to remove the end sequence token."""
     bad_words: List[str] = []
     """List of tokens not allowed to be generated."""
-    top_p: int = 1
+    top_p: float = 1.0
```
Comment (chyroc) [inline libs/community/langchain_community/llms/nlpcloud.py:41]: https://docs.nlpcloud.com/#generation

image

File: libs/community/langchain_community/embeddings/ollama.py [hunk 3]
```diff
@@ -97,7 +97,7 @@ class OllamaEmbeddings(BaseModel, Embeddings):
     will give more diverse answers, while a lower value (e.g. 10)
     will be more conservative. (Default: 40)"""
 
-    top_p: Optional[int] = None
+    top_p: Optional[float] = None
```
Comment (chyroc) [inline libs/community/langchain_community/embeddings/ollama.py:100]: https://github.com/jmorganca/ollama/blob/main/docs/modelfile.md

image

**Comment by chyroc** [comment]
@hwchase17

**Review by baskaryan — APPROVED** [review]
thank you @chyroc!

**Pull request merged**